enterprise_access/apps/customer_billing/models.py

```diff
@@ -0,0 +1,160 @@
+"""
+Models for customer billing app.
+"""
+from datetime import timedelta
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core.validators import validate_slug
+from django.db import models, transaction
+from django.utils import timezone
+from django_extensions.db.models import TimeStampedModel
+from simple_history.models import HistoricalRecords
+
+User = get_user_model()
+
+
+class EnterpriseSlugReservation(TimeStampedModel):
+    """
+    Temporary reservation of enterprise customer slugs to prevent race conditions
+    during the checkout-to-provisioning flow.
+
+    Only one active reservation per user is allowed at a time.
+    """
+    class Meta:
+        verbose_name = "Enterprise Slug Reservation"
+        verbose_name_plural = "Enterprise Slug Reservations"
+
+    user = models.OneToOneField(
+        User,
+        on_delete=models.CASCADE,
+        related_name='enterprise_slug_reservation'
+    )
+    slug = models.SlugField(
+        max_length=255,
+        unique=True,
+        validators=[validate_slug],
+        help_text="Reserved enterprise customer slug"
+    )
+    expires_at = models.DateTimeField(
+        db_index=True,
+        help_text="Reservation expiration timestamp"
+    )
+    stripe_checkout_session_id = models.CharField(
+        db_index=True,
+        max_length=255,
+        blank=True,
+        null=True,
+        help_text="Associated Stripe checkout session ID"
+    )
+    history = HistoricalRecords()
+
+    def __str__(self):
+        return f"{self.user.email}, {self.slug} (expires {self.expires_at})"
+
+    def is_expired(self):
+        """Check if this reservation has expired."""
+        if self.expires_at:
+            return timezone.now() > self.expires_at
+        return None
+
+    @classmethod
+    def get_reservation_duration(cls):
+        """Get reservation duration from settings or default to 30 minutes."""
+        return getattr(settings, 'ENTERPRISE_SLUG_RESERVATION_MINUTES', 30)
```

Comment:
According to stripe's AI assistant: "When a Checkout Session is created, Stripe automatically sets the expires_at timestamp to 24 hours in the future." -- so maybe bump this default from 30 minutes to 24 hours?